composer/algorithms/copypaste/copypaste.py

```diff
@@ -0,0 +1,403 @@
+from composer.loggers import Logger
+
+log = logging.getLogger(__name__)
+
+__all__ = ['CopyPaste', 'copypaste_batch']
+
+_MAX_TORCH_SEED = 0xffff_ffff_ffff_ffff
+
+
+def copypaste_batch(images, masks, configs):
+    """
+    copy-paste is an augmentation method. Two images are randomly chosen (with 
+    replacement) from a given batch of images and their corresponding masks, i.e.,
+    the source and the target target. A number of instances from the source are
+    selected to be copied into the target. The number of copied instanes is always
+    less than a number determined by the minimum of ``max_copied_instance`` and 
+    total number of instances in the source. A set of instances are then randomly
+    chosen (without replacement) from the source. Each instance goes through a set
+    of transformation and jittering, e.g., horizontal flipping, rescaling, and 
+    cropping. The resulting "jittered" instance is then checked for its area and 
+    transformed instances with an area smaller than the configured threshold. 
+    The resulting jittered instances are then pasted on a random position on 
+    the target. Same procedure is also applied on the masks of the source and 
+    the target. If a copied instances mask collides with an exisiting mask on 
+    the target, the copied instance's mask overlays the original mask on the 
+    target.
+
+    Args:
+        input (torch.Tensor): input tensor of shape ``(N, C, H, W)``.
+        target (torch.Tensor): target tensor of shape ``(N, H, W)``.   
+        configs (dict): dictionary containing the configurable hyperparameters.    
+
+    Returns:
+        out_images (torch.Tensor): batch of images after copypaste has been
+            applied.
+        out_masks (torch.Tensor): batch of corresponding masks after applying 
+        copypaste to them
+
+    Example:
+        .. testcode::
+
+            import torch
+            from composer.functional import copypaste_batch
+
+            N, C, H, W = 2, 3, 4, 5
+            num_classes = 10
+            configs = {
+                "p": 1.0,
+                "max_copied_instances": None,
+                "area_threshold": 100,
+                "padding_factor": 0.5,
+                "jitter_scale": (0.01, 0.99),
+                "jitter_ratio": (1.0, 1.0),
+                "p_flip": 1.0,
+                "bg_color": 0
+            }
+            X = torch.randn(N, C, H, W)
+            y = torch.randint(num_classes, size=(N, H, W))
+            out_images, out_masks = cutmix_batch(X, y, configs)
+    """
+    batch_idx = 0
+    out_images = torch.zeros_like(images)
+    out_masks = torch.zeros_like(masks)
+
+    assert images.size(dim=0) == masks.size(
+        dim=0), "Number of images and masks in the batch do not match!"
+    batch_size = images.size(dim=0)
+
+    while batch_idx < batch_size:
+        [i, j] = np.random.randint(0, high=batch_size, size=2)
+
+        num_instances = _count_instances(masks[i])
```

Comment:
```suggestion
        num_instances = len(torch.unique(masks[i])) - 1
```
This can be probably inlined too?